lib/streamlit/runtime/scriptrunner/script_run_context.py

```diff
@@ -69,6 +69,10 @@ class ScriptRunContext:
     )
     script_requests: Optional[ScriptRequests] = None
 
+    _experimental_query_params_used = False
+    # TODO(willhuang1997): the variable name of this may change as new api may be experimental too
```

Comment:
TODOs are not great as we see in 1.28.1 release so I will add a checklist in `feature/st_query_params_2` to make sure this is checked off and removed before merge into develop.

Reply:
Changed this to `_production_query_params_used` as that seems to differentiate the `_experiemental_query_params_used` decently well but am open to other variable names